tests/browser_context_test.go

```diff
@@ -688,3 +689,156 @@ func TestBrowserContextTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestBrowserContextWaitForEvent(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name            string
+		event           string
+		optsOrPredicate *optsOrPredicate
+		wantErr         string
+	}{
+		{
+			name:  "successfully wait for page creation",
```

Comment:
No strong opinion on this one as I'll share what I find useful for me.

Should we keep these test names concise? So the test run "glancability" would be better. I also prefer using underscores in test names because it makes it easier to copy fast and find-search when they fail.

We can elaborate on what's going on with a comment.

```go
		{
			// successfully wait for page creation with only predicate
			name:            "with_predicate",
			event:           "page",
			optsOrPredicate: &optsOrPredicate{justPredicate: stringPtr("() => true;")},
		},
```

I omitted "page" from the name because the event is `"page"` is there. And I didn't say "wait" because the test is about wait for event.

Reply:
Sure, I can give it a go. I personally prefer to be a bit more explicit but i think being consistent is better. Resolved in b2d2503d7fa48f4800e38cf37308a9148bf36caa

Reply:
Explicitness is suitable for other cases, yeah, I agree. On the other hand, naming in Go is usually concise, including test names :) Thanks for giving it a shot ❤️